blivet/devicelibs/disk.py

```diff
@@ -0,0 +1,110 @@
+from collections import namedtuple
+
+from . import raid
+from .. import errors
+from .. import util
+from ..size import Size
+
+lsm = None
+
+_HBA_PLUGIN_URIS = ("hpsa://", "megaraid://")
+LSMInfo = namedtuple('HBAVolumeInfo', ['system', 'nodes', 'raid_type', 'raid_stripe_size', 'raid_disk_count'])
+""" .. class:: LSMInfo
+
+        .. attribute:: system (str): descriptive name of HBA unit
+        .. attribute:: nodes (list[str]): list of device node paths for the volume
+        .. attribute:: raid_type (:class:`~.devicelibs.raid.RAIDLevel` or None): RAID level
+        .. attribute:: raid_stripe_size (:class:`~.size.Size` or None): stripe size
+        .. attribute:: raid_disk_count (int or None): number of disks in the RAID set
+"""
+volumes = dict()
+_raid_levels = dict()
+
+
+class _LSMRAIDLevelStub(raid.RAIDLevel):
+    def __init__(self, name):
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def names(self):
+        return [self.name]
+
+    @property
+    def min_members(self):
+        return 0
+
+    def has_redundancy(self):
+        return False
+
+    def is_uniform(self):
+        return False
+
+
+class _LSMDependencyGuard(util.DependencyGuard):
+    error_msg = "libstoragemgmt functionality not available"
+
+    def _check_avail(self):
+        global lsm
+        try:
+            import lsm  # pylint: disable=redefined-outer-name
+        except ImportError:
+            lsm = None
+
+        return lsm is not None
+
+_lsm_required = _LSMDependencyGuard()
+
+
+def _update_lsm_raid_levels():
+    """ Build a mapping of lsm.RAID_TYPE->blivet.devicelibs.raid.RAIDLevel """
+    global _raid_levels
+    _raid_levels = dict()
+    lsm_raid_levels = dict((k, v) for (k, v) in lsm.Volume.__dict__.items() if k.startswith("RAID_TYPE_"))
```

Comment:
Yeah, I've just become fond of the explicit constructor calls -- especially for `dict` and `list`. I think when they introduced the `set` initialization syntax that looked like `dict`'s it made me want to be more explicit.